truffle/erc721-example/truffle-config.js

```diff
@@ -1,26 +1,9 @@
-const HDWalletProvider = require('truffle-hdwallet-provider')
+const HDWalletProvider = require('@truffle/hdwallet-provider')
 const NonceTrackerSubprovider = require('web3-provider-engine/subproviders/nonce-tracker')
 const utils = require('web3-utils')
-const MNEMONIC = process.env.MNEMONIC || process.env.MNEMONIC
+const MNEMONIC = process.env.MNEMONIC
 const startIndex = 0
-const numberOfAccounts = 3
-let hdWalletProvider
-
-const setupWallet = (
-    url
-) => {
-    if (!hdWalletProvider) {
```

Comment:
We probably need this, since it is also used for ropsten and testnet setup, also it can be more verbose when initiated via object.
  hdWalletProvider = new HDWalletProvider({
          mnemonic: MNEMONIC,
          providerOrUrl: url,
          startIndex: 0,
          numberOfAddresses: 3,
          shareNonce: true,
        });